### ISSUE TITLE
Password on share: enforce vs. ask

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/server#15459]
TODO
- [x] Android: https://github.com/nextcloud/android/pull/3989
- [x] Desktop
- [ ] iOS
- [x] server: NC17+

![image](https://user-images.githubusercontent.com/5836855/57435273-3a685b80-723d-11e9-9c46-9e594713c643.png)

We have two different options here
- enforce: password needs to be set
  -  the dialog should have no option to skip it
- always ask: ask for a password, but it is not required
  -  the dialog should indicate that it is ok to skip it

Enforce vs. optional
![2019-05-09-094108](https://user-images.githubusercontent.com/5836855/57435808-9d0e2700-723e-11e9-8a4c-c509c49dd12d.png) ![2019-05-09-124724](https://user-images.githubusercontent.com/5836855/57447985-a3110180-7258-11e9-9085-bd1379dff310.png)

@nextcloud/designers

**Comment by skjnldsv** [comment]
IS this a feature request? A bug? :thinking:

**Comment by tobiasKaminsky** [comment]
This is a meta-issue for clients, as indicated by labels.
We agreed to have those issues centrally in server and, if needed, then individual in each repo.
Once the client is working, this should be noted here and if every client/platform has included it, we can close this.

tl;dr: for feature parity we track this here with label "client"

**Comment by tobiasKaminsky** [comment]
In capability this is:

Enforce password:
```
"files_sharing": {
          "api_enabled": true,
          "public": {
            "enabled": true,
            "password": {
              "enforced": true
            },
```

But I do not see a capability for "always ask for password"?
@nextcloud/server-triage

**Comment by tobiasKaminsky** [comment]
I added a PR to support askForPassword capability.

**Comment by jancborchardt** [comment]
The mandatory/optional state of the password could also not only be indicated by the dialog, but by the flow. If passwords are enforced, the prompt to enter the password should directly show and say something like "Passwords are mandatory".

**Comment by tobiasKaminsky** [comment]
> the prompt to enter the password should directly show and say something like "Passwords are mandatory".

This is already the case. but in both cases. The only difference is the title…

**Comment by tobiasKaminsky** [comment]
So @jancborchardt are you fine with this? :-D

**Comment by jancborchardt** [comment]
Yes, sorry – I :+1:’d your reply and meant that I was fine with it. :D

**Comment by joshtrichards** [comment]
Is the final unchecked item for iOS addressed in nextcloud/ios#1804 (merged & released)? If so, I think this is closable. 

I'll leave this open for now until confirmed by one of you or I turn up better evidence in already merged PRs. :)
